### DISCUSSION TITLE
Finding when a link has been created and has been connected to a source and target

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#1758] [Q&A]
I have noticed that if I create a couple of elements and embed them inside another element and then I try to create a link between them, the link doesn't get `.reparent()-ed` automatically. (Notice: I'm talking about creating all these elements dragging and dropping the elements on the actual canvas; not programmatically)

I have a couple of questions:

1) Is there a way to automatically reparent links?
2) If "no" to the previos question, is there a way I can detect when a link has been created and it has been connected to a source and target elements.

I know I can listen to the graph's `add` event, but this will fire the event too soon (it will fire before the link has been connected to a target element). I tried to listen to `link:pointerup` and find, somehow, if it's a newly created link, but I couldn't figure out a way.

**Comment by kumilingus** [comment]
The question is if the `reparenting` should have been done automatically. Perhaps it's a bug.

For now, please listen to the paper `link:connect` event.

**Reply by alexandernst** [reply]
Great! Will do! Please let me know (when you get the chance to look into this) if you consider this a bug so I can write a comment in my code linking to this issue. 🙏
